### PR TITLE
batchName added to active registration for post response

### PR DESCRIPTION
**Pull request by jaserud**
reference issue: #86

### CODE REVIEW

File: src/resources/swagger.yaml [hunk 1]
```diff
@@ -55,7 +55,17 @@ paths:
                     type: string
                   code:
                     type: string
-                    enum: ["TSV_PARSING_FAILED"]
+                    enum: ["TSV_PARSING_FAILED", "INVALID_FILE_ERROR"]
```
Comment (blabadi) [inline src/resources/swagger.yaml]: you also need to add `batchName` to  the registration schema : #/components/schemas/Registration

File: src/resources/swagger.yaml [hunk 2]
```diff
@@ -55,7 +55,17 @@ paths:
...
+              examples:
+                TSVParseError:
+                  value:
+                    msg: "string"
+                    code: "TSV_PARSING_FAILED"
+                InvalidFIleNameError:
+                  value:
+                    msg: "string"
+                    code: "INVALID_FILE_ERROR"
```
Comment (blabadi) [inline src/resources/swagger.yaml]: change this to `INVALID_FILE_NAME`

File: test/integration/submission/submission.spec.ts [hunk 3]
```diff
@@ -396,6 +397,33 @@ describe("Submission Api", () => {
...
+        .post("/submission/program/ABCD-EF/registration")
+        .type("form")
+        .attach("registrationFile", file, "thisIsARegistration.tsv")
+        .auth(JWT_ABCDEF, { type: "bearer" })
+        .end(async (err: any, res: any) => {
+          try {
+            res.should.have.status(400);
+            res.body.should.deep.eq({
+              msg: "invalid file name, must be registration*.tsv",
+              code: "INVALID_FILE_NAME"
```
Comment (blabadi) [inline test/integration/submission/submission.spec.ts]: use the enum value

File: src/utils.ts [hunk 4]
```diff
@@ -88,6 +88,10 @@ export function isString(value: any): value is string {
   return value instanceof String;
 }
 
+export const isStringMatchRegex = (expression: string, value: string) => {
+  return RegExp(expression).test(value);
```
Comment (blabadi) [inline src/utils.ts]: could this throw an error?

Comment (jaserud) [reply src/utils.ts]: Yea, if the expression is not valid

**Pull request merged**